From 69f47fc2e86aed46fd58bf12430f1206be4f9c7e Mon Sep 17 00:00:00 2001
In-Reply-To: <xmqqint5l4i9.fsf@gitster.mtv.corp.google.com>
References: <xmqqint5l4i9.fsf@gitster.mtv.corp.google.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sun, 19 Mar 2017 14:32:17 +0000
Subject: [PATCH v2] push: Re-include "push.default=tracking" in the
 documentation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change the documentation for push.tracking=* to re-include a mention
of what "tracking" does.

The "tracking" option was renamed to "upstream" back in
53c4031 ("push.default: Rename 'tracking' to 'upstream'", 2011-02-16),
this section was then subsequently rewritten in 87a70e4 ("config doc:
rewrite push.default section", 2013-06-19) to remove any mention of
"tracking".

Maybe we should just warn or die nowadays if this option is in the
config, but I had some old config of mine use this option, I'd
forgotten that it was a synonym, and nothing in git's documentation
mentioned that.

That's bad, either we shouldn't support it at all, or we should
document what it does. This patch does the latter.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---

On Thu, Oct 6, 2016 at 8:24 PM, Junio C Hamano <gitster@pobox.com> wrote:
> Sorry, I wasn't paying attention to this thread.

And neither was I it would appear, and for a much longer time...

> It seems that 87a70e4ce8 ("config doc: rewrite push.default
> section", 2013-06-19) removed that mention by accident?  The log
> message of the commit does not say it actively wanted to remove
> mention of `tracking` and/or why it wanted to do so, so I agree that
> resurrecting that parenthetical mention is the easiest course of
> action at this point.
>
> However.
>
> With today's description of push.default choices, each of which is a
> full fledged paragraph, I no longer have the objection I had in
>
>     https://public-inbox.org/git/7vip6dgmx2.fsf@alter.siamese.dyndns.org/
>
> against having `tracking` as a separate bullet item.  If we add
>
>     * `tracking` - a deprecated synonym for `upstream`; do not use this.
>
> to today's list, it would stand out as something different from
> others and it will not cause the confusion I feared in the
> discussion we had in early 2013.  As Jonathan Nieder argued in the
> thread back then, having it as one of the bullet point would help
> people locate it without using "search" \C-s or / feature.

Makes senes. Here's a re-send that fixes this. I slighly changed the
wording you suggested to be consistent with the other existing cases
in config.txt:
    
    5 matches for "eprecated.*for" in buffer: config.txt
       2328:	This is a deprecated synonym for `repack.writeBitmaps`.
       2462:* `tracking` - This is a deprecated synonym for `upstream`.
       2813:	Deprecated alias for 'sendemail.smtpEncryption = ssl'.
       2853:	Deprecated alias for `sendemail.signedoffbycc`.
       3180:	Deprecated alias for `versionsort.suffix`.  Ignored if

 Documentation/config.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Documentation/config.txt b/Documentation/config.txt
index eccc012672..988659e16c 100644
--- a/Documentation/config.txt
+++ b/Documentation/config.txt
@@ -2459,6 +2459,8 @@ push.default::
   pushing to the same repository you would normally pull from
   (i.e. central workflow).
 
+* `tracking` - This is a deprecated synonym for `upstream`.
+
 * `simple` - in centralized workflow, work like `upstream` with an
   added safety to refuse to push if the upstream branch's name is
   different from the local one.
-- 
2.11.0

